diff -urwE '--exclude=*.patch' '--exclude=*.fix.java' /mnt/d4j_bug_info/buggyfiles/math/math_35_buggy/src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java /mnt/D4JPatches/Manual2023/patches_correct/Math/Math_35/Math_35_TBar_normal/src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java
--- /mnt/d4j_bug_info/buggyfiles/math/math_35_buggy/src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java	2022-11-20 04:09:32.000000000 -0800
+++ /mnt/D4JPatches/Manual2023/patches_correct/Math/Math_35/Math_35_TBar_normal/src/main/java/org/apache/commons/math3/genetics/ElitisticListPopulation.java	2023-11-13 22:37:06.000000000 -0800
@@ -62,7 +62,7 @@
      */
     public ElitisticListPopulation(final int populationLimit, final double elitismRate) {
         super(populationLimit);
-        this.elitismRate = elitismRate;
+        setElitismRate(elitismRate);
     }
 
     /**
Only in /mnt/D4JPatches/Manual2023/patches_correct/Math/Math_35/Math_35_TBar_normal/src/main/java/org/apache/commons/math3/genetics: ElitisticListPopulation.java.orig
