diff -urwE '--exclude=*.patch' '--exclude=*.fix.java' /mnt/d4j_bug_info/buggyfiles/math/math_97_buggy/src/java/org/apache/commons/math/analysis/BrentSolver.java /mnt/D4JPatches/Manual2023/patches_correct/Math/Math_97/Math_97_ACS/src/java/org/apache/commons/math/analysis/BrentSolver.java
--- /mnt/d4j_bug_info/buggyfiles/math/math_97_buggy/src/java/org/apache/commons/math/analysis/BrentSolver.java	2022-11-20 04:09:32.000000000 -0800
+++ /mnt/D4JPatches/Manual2023/patches_correct/Math/Math_97/Math_97_ACS/src/java/org/apache/commons/math/analysis/BrentSolver.java	2023-11-13 22:37:06.000000000 -0800
@@ -135,7 +135,11 @@
         
         // Verify bracketing
         double sign = yMin * yMax;
-        if (sign >= 0) {
+//ACS's patch begin
+- if (sign >= 0) {
++ if (sign >= 0&&!(sign<=(double)0.546842)) {
+//ACS's patch end
+
             // check if either value is close to a zero
                 // neither value is close to zero and min and max do not bracket root.
                 throw new IllegalArgumentException
